70_FLIPFluids/main.cpp

```diff
@@ -0,0 +1,2240 @@
+#include <nabla.h>
+
+#include "nbl/application_templates/MonoAssetManagerAndBuiltinResourceApplication.hpp"
+#include "../common/SimpleWindowedApplication.hpp"
+#include "../common/InputSystem.hpp"
+#include "../common/Camera.hpp"
+
+#include "gpuRadixSort.h"
+
+#include "glm/glm/glm.hpp"
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/cpp_compat/matrix.hlsl>
+
+#include "app_resources/common.hlsl"
+#include "app_resources/descriptor_bindings.hlsl"
+
+using namespace nbl::hlsl;
+using namespace nbl;
+using namespace core;
+using namespace hlsl;
+using namespace system;
+using namespace asset;
+using namespace ui;
+using namespace video;
+
+
+enum SimPresets
+{
+	CENTER_DROP,
+	LONG_BOX
+};
+
+struct Particle
+{
+	float32_t4 position;
+    float32_t4 velocity;
+
+    uint32_t id;
+    uint32_t pad[3];
+};
+
+struct SGridData
+{
+    float gridCellSize;
+    float gridInvCellSize;
+    float pad0[2];
+
+    int32_t4 particleInitMin;
+    int32_t4 particleInitMax;
+    int32_t4 particleInitSize;
+
+    float32_t4 worldMin;
+    float32_t4 worldMax;
+    int32_t4 gridSize;
+};
+
+struct SMVPParams
+{
+	float cameraPosition[4];
+
+	float MVP[4*4];
+	float M[4*4];
+    float V[4*4];
+	float P[4*4];
+};
+
+struct SParticleRenderParams
+{
+    float radius;
+    float zNear;
+    float zFar;
+	float pad;
+};
+
+struct VertexInfo
+{
+	float32_t4 position;
+	float32_t4 vsSpherePos;
+
+    float radius;
+    float pad;
+
+    float32_t4 color;
+	float32_t2 uv;
+};
```

Comment:
the whole point of sharing a header between C++ and HLSL is that you can share struct definitions

and the only way thats possible if you use sized int/float/uint with the `_t` sufficies (we only made these available in C++ to discourage the use of all the other types)